src/hooks/useShortcut.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useCallback, useReducer } from 'react'
+
+const keysReducer = (state, action) => {
+  switch (action.type) {
+    case 'set-key-down':
+      const keydownState = { ...state, [action.key]: true }
+      return keydownState
+    case 'set-key-up':
+      const keyUpState = { ...state, [action.key]: false }
+      return keyUpState
+    case 'reset-keys':
+      const resetState = { ...action.data }
+      return resetState
+    default:
+      return state
+  }
+}
+
+const useShortcut = (shortcutKeys, callback) => {
+  if (!Array.isArray(shortcutKeys))
+    throw new Error(
+      'The first parameter to `useKeyboardShortcut` must be an ordered array of `KeyboardEvent.key` strings.'
+    )
+
+  if (!shortcutKeys.length)
+    throw new Error(
+      'The first parameter to `useKeyboardShortcut` must contain atleast one `KeyboardEvent.key` string.'
+    )
+
+  if (!callback || typeof callback !== 'function')
+    throw new Error(
+      'The second parameter to `useKeyboardShortcut` must be a function that will be envoked when the keys are pressed.'
+    )
+
+  const initalKeyMapping = shortcutKeys.reduce((currentKeys, key) => {
+    currentKeys[key.toLowerCase()] = false
+    return currentKeys
+  }, {})
+
+  const [keys, setKeys] = useReducer(keysReducer, initalKeyMapping)
+
+  const keydownListener = useCallback(
```

Comment:
Did you run into problems before using `useCallback`? I wonder why you need it.

Reply:
I followed a tutorial and liked the idea that useShortcut is not rendered permanently when the button is pressed. 
But it runs fine without the useCallback.

Reply:
_In order to prevent this function being recreated on every update of this hook, we will wrap the function in useCallback which will only recreate the function if the shortcutKeys or the keys objects have changed._  – from the tutorial-